### PR TITLE
Spark: Support parallel delete files when abort

### PR DESCRIPTION
**Pull request by ConeyLiu**
This patch adds support for parallel delete files when commit aborting.

### CODE REVIEW

File: spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java [hunk 1]
```diff
@@ -43,4 +43,8 @@ private SparkSQLProperties() {
   // Controls whether to check the order of fields during writes
   public static final String CHECK_ORDERING = "spark.sql.iceberg.check-ordering";
   public static final boolean CHECK_ORDERING_DEFAULT = true;
+
+  // Control whether to parallel delete files when abort
+  public static final String DELETE_FILES_PARALLEL_WHEN_ABORT = "spark.sql.iceberg.delete-files-parallel-when-abort";
```
Comment (rdblue) [inline spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java:48]: The name "delete-files-parallel-when-abort" is very long. How about "parallel-abort-enabled"?

Does this even need to be an option? Why would anyone turn it off?

File: spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java [hunk 2]
```diff
@@ -204,6 +205,7 @@ private void commitOperation(SnapshotUpdate<?> operation, String description) {
   private void abort(WriterCommitMessage[] messages) {
     Map<String, String> props = table.properties();
     Tasks.foreach(files(messages))
+        .executeWith(writeConf.deleteFilesParallelWhenAbort() ? ThreadPools.getWorkerPool() : null)
```
[the remark this answers is not in the document]
Comment (rdblue) [reply spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java:208]: I'd probably use the worker pool for now. No need to extend it until someone needs it.

File: spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java (hunk 1 above)
Comment (ConeyLiu) [reply spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java]: @rdblue I remove the option. Previously I want to have the ability to keep the behavior as before.

**Comment by ConeyLiu** [comment]
Thanks, @singhpk234 @rdblue for the reviewing, the code has been updated. Please take another look.

File: spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java [hunk 3]
```diff
@@ -213,6 +214,7 @@ private void abort(WriterCommitMessage[] messages) {
     if (cleanupOnAbort) {
       Map<String, String> props = table.properties();
       Tasks.foreach(files(messages))
+          .executeWith(ThreadPools.getWorkerPool())
```
Comment (rdblue) [inline spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java:217]: @RussellSpitzer, what do you think about this change?

Comment (RussellSpitzer) [reply spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java:217]: Seems fine to me, although I really wish we had a more discrete pool for things like this. I find that our worker pool is hard to configure for end users.

**Review by RussellSpitzer — APPROVED** [review]

File: spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java (hunk 3 above)
Comment (rdblue) [reply spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java:217]: Should we add a Spark-managed pool? We have the ability to plug in the pool used in most places now, thanks to @yittg.

**Comment by rdblue** [comment]
Thanks, @ConeyLiu!

**Comment by ConeyLiu** [comment]
Thanks all for the review.

**Pull request merged**